### ISSUE TITLE
(Proposal) A cookbook style flavour for ziglearn

### ISSUE THREAD

**Issue by timmyjose** [Sobeston/ziglearn#22]
First of all, thanks for creating this resource. This has been personally useful for me! 

In fact, as I was studying some of the snippets (readers/writers, for instance), I realised that this fits in very well with a cookbook style of approach. The canonical example I can give is the wonderful [Lisp cookbook](https://lispcookbook.github.io/cl-cookbook/). 

I know (and agree with the fact) that "cookbook" seems to have a very particular connotation in modern usage, but I feel that that does not detract too much from the overall idea. To illustrate, from the same link, a [sample chapter on threads](https://lispcookbook.github.io/cl-cookbook/process.html). 

Perhaps you could consider this cookbook approach since the chapters themselves already contain very useful (and runnable!) example code. I believe quite strongly that such a resource will become indispensable very soon.

**Comment by timmyjose** [comment]
Not apropos, but the part on implementing a custom `Writer` was super useful for me. Thanks!

**Comment by schroffl** [comment]
I've read the term "Cookbook" a lot of times before, but I just noticed that I'm not exactly sure what it means.
Does it just contain snippets and simple explanations for common things? Like "Loops, iteration, mapping" in the LISP example you posted?

If so, I love the idea. When starting out with Zig there were a lot of things for me where I had to look at the source to get a grasp. Having a basic overview would be awesome.

**Comment by Sobeston** [comment]
I intend things to be largely dominated by examples, with longer (albeit concise) examples in chapter 2 similar to the writer implementation one. I'm not sure what kinds of changes would be made here

**Comment by timmyjose** [comment]
> Does it just contain snippets and simple explanations for common things? Like "Loops, iteration, mapping" in the LISP example you posted?

Yes! That's exactly what it is. Some examples may have some extended explanations, but all within the same context. 

The contents you have are already more or less along the same lines. Perhaps just the table of contents et al would need a bit of tweaking to find the relevant sections more easily.
